Feature/16 App Bar

khaledeid1k opened the pull request.

khaledeid1k committed:
Add MovieAppBar component

This commit introduces a new Composable function, `MovieAppBar`, to the design system.

The `MovieAppBar` displays:
- A back button
- The CineVerse logo
- A title ("CineVerse") and a caption ("Movie App")
- An add button

It also includes a horizontal divider at the bottom.
A preview function, `MovieAppBarPreview`, is provided to showcase the component.

khaledeid1k committed:
feat: Add new string resources

This commit introduces new string resources to the `strings.xml` file.

The following strings have been added:
- `caption`: "Caption"
- `title`: "Title"

khaledeid1k committed:
Merge branch 'develop' into feature/15-app-bar

Zeinab979 commented on design_system/src/main/java/com/moscow/cineverse/designSystem/component/MovieAppBar.kt:
```diff
@@ -0,0 +1,91 @@
+import com.moscow.cineverse.designSystem.theme.ThemeState
+
+@Composable
+fun MovieAppBar(modifier: Modifier = Modifier,
```
make it more generic and reusable across multiple screens

AdelTamer35 replied:
agree with you

AdelTamer35 reviewed (changes requested):
change develop to feature/2-component-general

ZeyadAbdullah679 commented on design_system/src/main/java/com/moscow/cineverse/designSystem/component/MovieAppBar.kt:
```diff
@@ -0,0 +1,91 @@
+        )
+    }
+}
+@Preview(showBackground = true)
```
make a preview for arabic and english and dark mode and light mode

ZeyadAbdullah679 commented on design_system/src/main/java/com/moscow/cineverse/designSystem/component/MovieAppBar.kt:
```diff
@@ -0,0 +1,91 @@
+                )
+            }
+            IconButton(
+                onClick = { addButtonClick() },
```
you can replace that with onClick = addButtonClick

ZeyadAbdullah679 commented:
change the pull request from the `develop` to `feature/2-component-general`

ZeyadAbdullah679 reviewed (changes requested):
redirect the pr to feature/2-component-general

khaledeid1k committed:
Refactor: Improve MovieAppBar and add multi-preview support

This commit refactors the `MovieAppBar` composable for better flexibility and introduces a `CineVersePreviews` annotation for easier multi-configuration previews.

**MovieAppBar Changes:**
- Increased logo size from 24dp to 32dp.
- Added parameters to control the visibility of the back button, add button, logo, and divider.
- Added parameters for optional title and caption text.
- Removed hardcoded string resources for title and caption.
- Updated preview to showcase the new parameters.

**Other Changes:**
- Enabled `autoMirrored` for the `outline_arrow_left` drawable.
- Created `CineVersePreviews` annotation to simplify defining multiple preview configurations (English/Arabic, with/without background).
- Removed unused string resources.

khaledeid1k committed:
Merge branch 'develop' into feature/15-app-bar
